File: apitechsec/src/main/java/com/epiis/apitechsec/dto/response/ResponseQuoteGetAll.java
package com.epiis.apitechsec.dto.response;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

import com.epiis.apitechsec.generic.ResponseGeneric;
import lombok.Getter;
import lombok.Setter;

public class ResponseQuoteGetAll extends ResponseGeneric {
	public List<ItemQuote> listQuote = new ArrayList<>();

	@Getter
	@Setter
	public static class ItemQuoteDetail {
		public Long id;
		public Long idProduct;
		public String productName;
		public Integer quantity;
		public BigDecimal unitPrice;
		public BigDecimal subtotal;
	}

	@Getter
	@Setter
	public static class ItemQuote {
		public Long id;
		public Long idClient;
		public String clientName;
		public String clientDocument;
		public BigDecimal total;
		public String status;
		public String notes;
		public String createdAt;
		public List<ItemQuoteDetail> listDetail = new ArrayList<>();
	}
}

File: techsec-frontend/src/app/api/models/response-quote-get-all.ts
import { ResponseProductGetAll } from './response-product-get-all';
import { ResponseUserGetAll } from './response-user-get-all';

export interface ResponseQuoteDetailGetAll {
  id: number;
  product: ResponseProductGetAll;
  quantity: number;
  unitPrice: number;
  subtotal: number;
}

export interface ResponseQuoteGetAll {
  id: number;
  client: ResponseUserGetAll;
  details: ResponseQuoteDetailGetAll[];
  total: number;
  createdAt: string;
  notes?: string;
}
